nss-fumbling-penguins/nutshell
# Rm tasks

## Link to Ticket
Closes # [7]()

## Description of Proposed Changes

Added the basic functionality for task management.

## Steps to Test

Outline the steps to test

```sh
git fetch --all
git checkout <feature_branch>

```
1. make sure you are logged in as any user
1. add tasks
1. check to make sure they are in database
1. check the available buttons to edit, permanently delete, and mark complete the tasks.
1. when edit is clicked, a modal should appear letting you edit the task which will update dom and database.
1. when complete is clicked, the buttons should all disappear and be replaced by a green complete marker. on page reload that task should no longer be loaded, but is still in the database.
1. when delete is clicked. the task should immediately disappear from the dom and database

## Impacted Areas in Application

List general components of the application that this PR will affect:

API

## Mentions @username

Tag users that need to review this code

## Definition of Done

1. The project must be fully documented. This includes the following:
    1. Complete README that documents the steps to install the code, how to install any dependencies, or system configuration needed.
    1. Every class must be documented with purpose, author, and methods.
    1. Every method must be documented with purpose and argument list - which itself must contain a short purpose for each argument.
1. The project must be able to run fully, and without errors, on each teammate's system.
1. Fulfills every requirement.
Every line of code has been peer reviewed.
